File: UESTC/2327.cpp
#include <bits/stdc++.h>
using namespace std;

const int Z1=10100;
const int Z2=100;

int n,a,b,x,vx,vy;
int cnt[Z1<<1|1],s[Z1<<1|1][Z2<<1|1];
long long ans;

int main()
{
	scanf("%d %d %d",&n,&a,&b);
	for (int i=1;i<=n;i++)
	{
		scanf("%d %d %d",&x,&vx,&vy);
		ans+=cnt[a*vx-vy+Z1]-s[a*vx-vy+Z1][vx+Z2];
		++cnt[a*vx-vy+Z1];
		++s[a*vx-vy+Z1][vx+Z2];
	}
	printf("%lld\n",ans<<1);
	return 0;
}

File: UESTC/2327.java
import java.util.*;

public class Main {
	static final int Z1 = 10100;
	static final int Z2 = 100;

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int n = in.nextInt();
		int a = in.nextInt();
		int b = in.nextInt();
		int[] cnt = new int[Z1 << 1 | 1];
		int[][] s = new int[Z1 << 1 | 1][Z2 << 1 | 1];
		long ans = 0;
		for (int i = 1; i <= n; i++) {
			int x = in.nextInt();
			int vx = in.nextInt();
			int vy = in.nextInt();
			ans += cnt[a * vx - vy + Z1] - s[a * vx - vy + Z1][vx + Z2];
			++cnt[a * vx - vy + Z1];
			++s[a * vx - vy + Z1][vx + Z2];
		}
		System.out.println(ans << 1);
	}
}